packages/itwin/tree-widget/src/components/trees/stateless/models-tree/internal/AlwaysAndNeverDrawnElementInfo.ts

```diff
@@ -0,0 +1,132 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { Observable, Subscription } from "rxjs";
+import {
+  BehaviorSubject, debounceTime, EMPTY, first, fromEventPattern, map, reduce, share, startWith, Subject, switchMap, takeUntil, tap,
+} from "rxjs";
+import { pushToMap } from "../../../models-tree/Utils";
+
+import type { Viewport } from "@itwin/core-frontend";
+import type { ElementsByParentQueryProps, ElementsQueryProps, ModelsTreeQueryHandler } from "./ModelsTreeQueryHandler";
+import type { BeEvent, Id64Set, Id64String, IDisposable } from "@itwin/core-bentley";
+const SET_CHANGE_DEBOUNCE_TIME = 20;
+
+interface CacheEntry {
+  byModel: Map<Id64String, Id64Set>;
+  byCategory: Map<string, Id64Set>;
+}
```

Comment:
Are we ever looking up entries by only category id? If not, I think this entry should be `Map<Id64String, Map<Id64String, Id64Set>>`, where outer map key is model id and inner map key is category id

Reply:
Made suggested change